File: Raven.Client.Java/lib/src/main/java/raven/client/connection/ReplicationInformerLocalCache.java
package raven.client.connection;

import java.io.File;

import org.apache.commons.io.FileUtils;
import org.apache.commons.lang.StringUtils;

import raven.abstractions.data.JsonDocument;
import raven.abstractions.json.linq.RavenJObject;
import raven.abstractions.logging.ILog;
import raven.abstractions.logging.LogManager;

public class ReplicationInformerLocalCache {

  private static ILog log = LogManager.getCurrentClassLogger();

  private static String tempDir = System.getProperty("java.io.tmpdir");

  public static JsonDocument tryLoadReplicationInformationFromLocalCache(String serverHash) {
    JsonDocument result = null;
    try {
      String path = "RavenDB Replication Information For - " + serverHash;
      File file = new File(tempDir, path);
      String fileContent = FileUtils.readFileToString(file);
      if (StringUtils.isBlank(fileContent)) {
        return null;
      }
      result = SerializationHelper.toJsonDocument(RavenJObject.parse(fileContent));
    } catch (Exception e) {
      log.error("Could not understand the persisted replication information", e);
      return null;
    }
    return result;
  }

  public static void trySavingReplicationInformationToLocalCache(String serverHash, JsonDocument document) {
    try {
      String path = "RavenDB Replication Information For - " + serverHash;
      File file = new File(tempDir, path);
      FileUtils.writeStringToFile(file, document.toJson().toString());
    } catch (Exception e) {
      log.error("Could not persist the replication information", e);
    }
  }

}

File: Raven.Client.Lightweight/Connection/ReplicationInformerLocalCache.cs
using System;
using System.Collections.Generic;
using System.IO;
#if !DNXCORE50
using System.IO.IsolatedStorage;
#endif
using Raven.Abstractions.Data;
using Raven.Abstractions.Extensions;
using Raven.Abstractions.Logging;
using Raven.Json.Linq;

namespace Raven.Client.Connection
{
    public static class ReplicationInformerLocalCache
    {
#if !DNXCORE50
        private readonly static ILog Log = LogManager.GetCurrentClassLogger();
#else
        private readonly static ILog Log = LogManager.GetLogger(typeof(ReplicationInformerLocalCache));
#endif

#if !DNXCORE50
        public static IsolatedStorageFile GetIsolatedStorageFile()
        {
#if MONO
            return IsolatedStorageFile.GetUserStoreForApplication();
#else
            return IsolatedStorageFile.GetMachineStoreForDomain();
#endif
        }
#endif

        public static void ClearReplicationInformationFromLocalCache(string serverHash)
        {
#if !DNXCORE50
            try
            {
                using (var machineStoreForApplication = GetIsolatedStorageFile())
                {
                    var path = "RavenDB 3.5 Replication Information For - " + serverHash;

                    if (machineStoreForApplication.GetFileNames(path).Length == 0)
                        return;

                    machineStoreForApplication.DeleteFile(path);
                }
            }
            catch (Exception e)
            {
                Log.ErrorException("Could not clear the persisted replication information", e);
            }
#endif
        }

        public static JsonDocument TryLoadReplicationInformationFromLocalCache(string serverHash)
        {
#if !DNXCORE50
            try
            {
                using (var machineStoreForApplication = GetIsolatedStorageFile())
                {
                    var path = "RavenDB 3.5 Replication Information For - " + serverHash;

                    if (machineStoreForApplication.GetFileNames(path).Length == 0)
                        return null;

                    using (var stream = new IsolatedStorageFileStream(path, FileMode.Open, machineStoreForApplication))
                    {
                        return stream.ToJObject().ToJsonDocument();
                    }
                }
            }
            catch (Exception e)
            {
                Log.ErrorException("Could not understand the persisted replication information", e);
                return null;
            }
#else
            return null;
#endif
        }

        public static void TrySavingReplicationInformationToLocalCache(string serverHash, JsonDocument document)
        {
#if !DNXCORE50
            try
            {
                using (var machineStoreForApplication = GetIsolatedStorageFile())
                {
                    var path = "RavenDB 3.5 Replication Information For - " + serverHash;
                    using (var stream = new IsolatedStorageFileStream(path, FileMode.Create, machineStoreForApplication))
                    {
                        document.ToJson().WriteTo(stream);
                    }
                }
            }
            catch (Exception e)
            {
                Log.ErrorException("Could not persist the replication information", e);
            }
#endif
        }

        public static void ClearLocalCache(string serverHash)
        {
#if !DNXCORE50

            using (var machineStoreForApplication = GetIsolatedStorageFile())
            {
                var path = "RavenDB Cluster Nodes For - " + serverHash;
                if (machineStoreForApplication.GetFileNames(path).Length == 0)
                    return;

                try
                {
                    machineStoreForApplication.DeleteFile(path);
                }
                catch (Exception)
                {
                    // not important
                }
            }
#endif
        }
    }
}
